core/provider/service.go

```diff
@@ -276,6 +277,31 @@ func (s *Service) RevokeAccess(a *domain.Appeal) error {
 	// handle the resolution for the appeal status
 }
 
+func (s *Service) Delete(id string) error {
+	p, err := s.providerRepository.GetByID(id)
+	if err != nil {
+		return fmt.Errorf("getting provider details: %w", err)
+	}
+
+	resources, err := s.resourceService.Find(map[string]interface{}{
+		"provider_type": p.Type,
+		"provider_urn":  p.URN,
+	})
+	if err != nil {
+		return fmt.Errorf("retrieving related resources: %w", err)
+	}
```

Comment:
will handle this in grpc handler

Reply:
just check that resource.Find won't return resouce.Notfound since it's a List method, not GetOne. Therefore, if the returned resources are empty, that's fine, shouldn't be considered as resource not found